fix for off-by-one error in encoder data

dougollerenshaw opened the pull request:
addresses issue #1473. Trims encoder values if length is off by 1 and prints a warning. Throws assertion error if length mismatch is greater than 1.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AllenInstitute/AllenSDK?pullRequest=1474) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/AllenInstitute/AllenSDK?pullRequest=1474) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AllenInstitute/AllenSDK?pullRequest=1474) it.</sub>

kschelonka commented:
Thank you for your contribution, Doug. If I understand correctly, this error is caused by the data rather than indicating a code bug. If that's the case, can you please update the `assert` statement to instead `raise` an exception if the data do not conform to expectations?

dougollerenshaw commented:
Thanks @kschelonka. I just made the change. Is that what you had in mind?

kschelonka commented:
@dougollerenshaw can you please sign the CLA so I can merge this? Thanks!

dougollerenshaw commented:
Hi @kschelonka I'm pretty sure I'd already done that, but I just tried again. I got some weird behavior where after clicking the confirm button, it brought me back to the same page.

The CLA notice above looks like this now (I wish I'd paid closer attention to what it looked like when I first came here a few minutes ago!):
![image](https://user-images.githubusercontent.com/19944442/80132830-218dae80-8551-11ea-826d-61f45beb3cd8.png)

Note the text: "You have signed the CLA already but the status is still pending? Let us recheck it."

When I click through, I see this:
![image](https://user-images.githubusercontent.com/19944442/80132913-42ee9a80-8551-11ea-9dc2-a3d1f28bd762.png)

Is it showing up as signed for you, or still pending?

kschelonka commented:
Huh, I don't know... It's not showing as signed to me. Were you logged into your github account when you clicked it?

dougollerenshaw commented:
@kschelonka yes, I was signed in earier. 

I just clicked through again and this is what I see:
![image](https://user-images.githubusercontent.com/19944442/80158254-e6a17000-857c-11ea-963a-ed163bf0c777.png)
So apparently it hadn't logged my earlier attempt to sign.

When I clicked "sign in with github to agree", I saw this briefly before getting redirected back to this page:
![image](https://user-images.githubusercontent.com/19944442/80158363-28321b00-857d-11ea-87af-ca35aed1d278.png)

And now when I click through again, I see the page I showed earlier that confirms that I've signed:
![image](https://user-images.githubusercontent.com/19944442/80158420-48fa7080-857d-11ea-97ba-d9a6cd97ffd1.png)

kschelonka commented:
Hmm, I'm still not seeing it as signed. Could you make an issue on the CLA repo with the steps to reproduce/screenshots?

dougollerenshaw commented:
@kschelonka: Done: https://github.com/cla-assistant/cla-assistant/issues/563

dougollerenshaw commented:
@kschelonka I'm just now noticing that the username listed for me in the CLA Assistant box above is 'dollerenshaw'. But my github username is 'dougollerenshaw'. 

Also, the commit record shows up as 'dollerenshaw':
![image](https://user-images.githubusercontent.com/19944442/80520245-06021980-893e-11ea-8c0b-bfc91af55084.png)

I wonder if I somehow didn't configure my gitkraken client (which I used to manage this PR) properly?

dougollerenshaw commented:
@kschelonka I solved it! It turned out that I had a nearly 7 year old github account associated with my institute email address (username @dollerenshaw) that I had completely forgotten about. Even though I was logged into my primary account (@dougollerenshaw) when committing here, github appears to have been matching my email up to that other account and attributing my commits to that username. 

I just logged in as that alternate user to sign the CLA. 

I also removed my institute address from that defunct account and added it to my primary github account, so hopefully all future commits will get associated with my account properly.

Sorry for the trouble!

dougollerenshaw commented:
And look: the commit history has been magically revised to show my correct username as well!
![image](https://user-images.githubusercontent.com/19944442/80524027-d48c4c80-8943-11ea-9879-fc189a933c62.png)

kschelonka commented:
@dougollerenshaw Nice debugging, thanks for getting this all fixed!
